### PR TITLE
docs: initial Sidetree Open API

### PR DESCRIPTION
**Pull request by troyronda**
Signed-off-by: Troy Ronda <troy.ronda@securekey.com>

### CODE REVIEW

File: docs/api/api.yml [hunk 1]
```diff
@@ -0,0 +1,594 @@
...
+              - $ref: '#/components/schemas/SidetreeRecoverOperation'
+              - $ref: '#/components/schemas/SidetreeDeactivateOperation'
+      responses:
+        '200':
+          description: Success
+        '400':
+          description: Bad Request
+        '401':
+          description: Unauthorized
+        '404':
```
Comment (sandrask) [inline docs/api/api.yml:41]: we return 410 for deactivate (gone)

File: docs/api/api.yml [hunk 2]
```diff
@@ -0,0 +1,594 @@
...
...aFZWUTBlVUVpTENKeVpXTnZkbVZ5ZVV0bGVTSTZleUp3ZFdKc2FXTkxaWGxJWlhnaU9pSXdNalE0WkRWaFlUbGxZamxqWVdZNE5EWmhNalZoTkRReE1qbGlPR013TURBek9HUTFObVJsTlROaVptTTNZbUU1TkRneU1tRTFNV1ZpTUdabU1EazNNbU1pZlN3aWJtVjRkRkpsWTI5MlpYSjVUM1J3U0dGemFDSTZJa1ZwUTI5aU5YZFZkMEV5U0VObVRGUjZjbmRHZG14b2JVSm5TRnB0ZEVsZmRXVXhNa1JuWHpsVlkxOXdlR2NpZlEiLCJvcGVyYXRpb25EYXRhIjoiZXlKdVpYaDBWWEJrWVhSbFQzUndTR0Z6YUNJNklrVnBSSEZMVDJ0ZlRsZHVZMkZrT0RJelYySm9WVGwyZUVwcmQwVnVTVFZHUlVNeU0xbDViRE5rUlZnNWJtY2lMQ0prYjJOMWJXVnVkQ0k2ZXlKQVkyOXVkR1Y0ZENJNkltaDBkSEJ6T2k4dmR6TnBaQzV2Y21jdlpHbGtMM1l4SWl3aWNIVmliR2xqUzJWNUlqcGJleUpwWkNJNklpTnphV2R1YVc1blMyVjVJaXdpZEhsd1pTSTZJbE5sWTNBeU5UWnJNVlpsY21sbWFXTmhkR2x2Ymt0bGVUSXdNVGdpTENKMWMyRm5aU0k2SW5OcFoyNXBibWNpTENKd2RXSnNhV05MWlhsSVpYZ2lPaUl3TTJKa01HVTBOREF3TlRKaU9UUXlaVE13T0dJNVptUXdPR1JpTWpsaFltTTRaRFl6TmpZNE5ESXpNMkZsT0Raa09Ea3lZVEk1WmpCak5qRTJabVV3TldVaWZWMHNJbk5sY25acFkyVWlPbHQ3SW1sa0lqb2lTV1JsYm5ScGRIbElkV0lpTENKMGVYQmxJam9pU1dSbGJuUnBkSGxJZFdJaUxDSnpaWEoyYVdObFJXNWtjRzlwYm5RaU9uc2lRR052Ym5SbGVIUWlPaUp6WTJobGJXRXVhV1JsYm5ScGRIa3VabTkxYm1SaGRHbHZiaTlvZFdJaUxDSkFkSGx3WlNJNklsVnpaWEpUWlhKMmFXTmxSVzVrY0c5cGJuUWlMQ0pwYm5OMFlXNWpaWE1pT2xzaVpHbGtPbk5wWkdWMGNtVmxPblpoYkhWbE1DSmRmWDFkZlgwIn0"
```
Comment (sandrask) [inline docs/api/api.yml:58]: decoding this results in  {"type":"create","suffixData"...  which is not our latest request

Comment (troyronda) [reply docs/api/api.yml:58]: @sandrask could you please provide an updated example?

File: docs/api/api.yml (hunk 1 above)
Comment (troyronda) [reply docs/api/api.yml:41]: will add.

Comment (troyronda) [reply docs/api/api.yml:41]: done

File: docs/api/api.yml [hunk 3]
```diff
@@ -0,0 +1,595 @@
...
+          description: Not Found
+        '500':
+          description: Server Error
+  /blockchain/transactions:
+    get:
+      tags:
+        - Blockchain
+      summary: Fetches Sidetree transactions in chronological order.
+      operationId: getTransactions
+      description: "Note: The call may not to return all Sidetree transactions in one batch, in which case the caller can use the transaction number of the last transaction in the returned batch to fetch subsequent transactions."
```
Comment (bstasyszyn) [inline docs/api/api.yml:134]: ... may not return ...

File: docs/api/api.yml [hunk 4]
```diff
@@ -0,0 +1,595 @@
...
+              properties:
+                code: 
+                  type: string
+                  const: "content_exceeds_maximum_allowed_size"
+                  description:  Content exceeds maximum allowed size
+            - type: object
+              properties:
+                code: 
+                  type: string
+                  const: "content_not_a_file"
```
Comment (bstasyszyn) [inline docs/api/api.yml:396]: We don't return this

File: docs/api/api.yml [hunk 5]
```diff
@@ -0,0 +1,595 @@
...
+                  description:  Content hash is invalid
+        '401':
+          description: Unauthorized
+        '404':
+          oneof:
+            - type: object
+              properties:
+                code: 
+                  type: string
+                  const: "value_time_lock_not_found"
```
Comment (bstasyszyn) [inline docs/api/api.yml:412]: We don't return this

File: docs/api/api.yml [hunk 6]
```diff
@@ -0,0 +1,595 @@
...
+                  description:  Lock not found.
+            - type: object
+              properties:
+                code: 
+                  type: string
+                  const: "value_time_lock_in_pending_state"
```
Comment (bstasyszyn) [inline docs/api/api.yml:418]: We don't return this

File: docs/api/api.yml (hunk 2 above)
Comment (sandrask) [reply docs/api/api.yml]: eyJkZWx0YV9oYXNoIjoiRWlCWE9wa2dGa1JuY0xETm9XQVdQT2lwSXBQMWxHS3Y4cmF5UFhCbWd1eC0yUSIsInJlY292ZXJ5X2NvbW1pdG1lbnQiOiJFaUNfTS1JSVAtMEs4V0dDeUlmbThZNjRZN1FwZjJ3U0x0cVN3QWtfSFlJRDh3IiwicmVjb3Zlcnlfa2V5Ijp7ImNydiI6IlAtMjU2Iiwia3R5IjoiRUMiLCJ4IjoiTmVKQ3lESU43NmZ6Z2NMMy0xRjBoOEFaa0tseDBqcmMwOXdOUTh0cUw5YyIsInkiOiJIWHpickRGRVNLT2tjcVJLbjhpamVkdDRRZEZ1bnhSUm1pM3ZGYmVrS093In19.eyJwYXRjaGVzIjpbeyJhY3Rpb24iOiJhZGQtcHVibGljLWtleXMiLCJwdWJsaWNfa2V5cyI6W3siaWQiOiJrZXkxIiwiandrIjp7ImNydiI6IlAtMjU2Iiwia3R5IjoiRUMiLCJ4IjoibUp4MFZPWnFLUDJ6Q3lRSno4OVJXRGxNTmc4X0djVnJBRlNnTUE4VHU3ayIsInkiOiJmbVFwcFZNa08tVnpJaGZTaVJBZjg4VkhQaTVqaHN4UEdtZ0I4amZGUGxVIn0sInR5cGUiOiJKd3NWZXJpZmljYXRpb25LZXkyMDIwIiwidXNhZ2UiOlsib3BzIl19LHsiaWQiOiJkdWFsLWF1dGgtZ2VuIiwiandrIjp7ImNydiI6IlAtMjU2Iiwia3R5IjoiRUMiLCJ4IjoiUGFwN1gyUEhWWW5tMFpTMEEtbWZaWV9tLVRwNmUzbEQ1TTJXMFBud0hUcyIsInkiOiJjVmt3ZlBDV3g5OFZ3UjBLcmJnUTFPTzNrQ3BFcGFMRTV4aGo0QVlzRXRjIn0sInR5cGUiOiJKd3NWZXJpZmljYXRpb25LZXkyMDIwIiwidXNhZ2UiOlsiYXV0aCIsImdlbmVyYWwiXX0seyJpZCI6ImR1YWwtYXNzZXJ0aW9uLWdlbiIsImp3ayI6eyJjcnYiOiJFZDI1NTE5Iiwia3R5IjoiT0tQIiwieCI6IlBJaWFyX2o4ZW5HMkEyTXVhN3I0MW5DbTJIOHFtY25Kd25Va0tpdGxYWlkiLCJ5IjoiIn0sInR5cGUiOiJFZDI1NTE5VmVyaWZpY2F0aW9uS2V5MjAxOCIsInVzYWdlIjpbImFzc2VydGlvbiIsImdlbmVyYWwiXX1dfSx7ImFjdGlvbiI6ImFkZC1zZXJ2aWNlLWVuZHBvaW50cyIsInNlcnZpY2VfZW5kcG9pbnRzIjpbeyJpZCI6Im9pZGMiLCJzZXJ2aWNlRW5kcG9pbnQiOiJodHRwczovL29wZW5pZC5leGFtcGxlLmNvbS8iLCJ0eXBlIjoiT3BlbklkQ29ubmVjdFZlcnNpb24xLjBTZXJ2aWNlIn0seyJpZCI6Imh1YiIsInNlcnZpY2VFbmRwb2ludCI6Imh0dHBzOi8vaHViLmV4YW1wbGUuY29tLy5pZGVudGl0eS9kaWQ6ZXhhbXBsZTowMTIzNDU2Nzg5YWJjZGVmLyIsInR5cGUiOiJIdWJTZXJ2aWNlIn1dfV0sInVwZGF0ZV9jb21taXRtZW50IjoiRWlDT3hBbEZ6Rjd4N0ZQZGRaQ1ZYaHJjZDJEZ2RGeVZJS1NKMm0xT21FZ0JSZyJ9

Comment (sandrask) [reply docs/api/api.yml]: @troyronda see above

File: docs/api/api.yml (hunk 6 above)
Comment (troyronda) [reply docs/api/api.yml]: done

File: docs/api/api.yml (hunk 5 above)
Comment (troyronda) [reply docs/api/api.yml]: done

File: docs/api/api.yml (hunk 4 above)
Comment (troyronda) [reply docs/api/api.yml]: done

File: docs/api/api.yml (hunk 3 above)
Comment (troyronda) [reply docs/api/api.yml]: done

File: docs/api/api.yml (hunk 2 above)
Comment (troyronda) [reply docs/api/api.yml]: will add examples in a followup.

**Pull request merged**